lib/libzpool/uzfs_io.c

```diff
@@ -109,6 +138,12 @@ uzfs_write_data(zvol_state_t *zv, char *buf, uint64_t offset, uint64_t len,
 		len -= bytes;
 	}
 exit_with_error:
+	if (zv->in_rebuilding_mode && is_rebuild && count)
+		goto chunk_io;
+
+	if (chunk_io)
+		umem_free(chunk_io, sizeof (*chunk_io));
```

Comment:
I think here too we need to use kmem_free.

Reply:
better do list_destroy also

Reply:
ok